front-end/src/app/header/header.component.ts

```diff
@@ -7,9 +8,28 @@ import { Component, OnInit } from '@angular/core';
 })
 export class HeaderComponent implements OnInit {
 
-  constructor() { }
+  private login: boolean;
+  private logout:boolean;
+  private tabs:boolean;
+
+  constructor(private router:Router) {
+    this.checkPath();
+    router.events.subscribe(() => {
+      this.checkPath();
+    });
+  }
 
   ngOnInit() {
   }
 
+  checkPath(): void{
```

Comment:
login/logout should be checked by if the local storage token exists, not the current url. Very unscalable.
I don't understand what tabs is controlling. Why would you want to hide the nav bar while logging in or in the home page?